scripts/builtin/hmm.dml

```diff
@@ -0,0 +1,36 @@
+#-------------------------------------------------------------
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+
+# This script implements the hidden markov model method
+# INPUT:
+# --------------------------------------------------------------------------------------------
+# X         Set of outputs
+# OUTPUT:
+# --------------------------------------------------------------------------------------------
+# P     Probability of the set of outputs
+# --------------------------------------------------------------------------------------------
+
+hmm = function(Matrix[Double] X) 
```

Comment:
since you are allowed to define more than one function in these files, we require you to mark the main function with: m_ , aka here it become m_hmm